src/main/java/com/avispl/symphony/dal/avdevices/power/surgex/defenderseries/SurgeXDefenderCommunicator.java

```diff
@@ -208,40 +261,64 @@ protected void authenticate() throws Exception {
 	 *
-	 * @param properties The {@link Properties} object to load the configuration into.
-	 * @throws ResourceNotReachableException if the properties file cannot be loaded.
+	 * @param properties The {@link Properties} object to populate with configuration values.
 	 */
 	private void loadProperties(Properties properties) {
 		try {
-			properties.load(getClass().getResourceAsStream("/application.properties"));
-		} catch (Exception e) {
-			throw new FileSystemNotFoundException(Constant.READ_PROPERTIES_FILE_FAILED);
+			properties.load(getClass().getResourceAsStream("/version.properties"));
+			properties.setProperty("adapter.uptime", String.valueOf(this.adapterInitializationTimestamp));
+		} catch (IOException e) {
+			this.logger.error(Constant.READ_PROPERTIES_FILE_FAILED + e.getMessage());
 		}
 	}
 
 	/**
-	 * Initializes device data by fetching the currentStatus and networkSettings.
+	 * Authenticates and attempts to fetch the device's current status and network settings.
+	 * Also extracts and sets the time zone info into the current status.
 	 *
-	 * @throws FailedLoginException If the login fails during data fetching.
+	 * @return {@code true} if data is successfully fetched and initialized; {@code false} otherwise.
+	 * @throws FailedLoginException if authentication fails.
 	 */
-	private void setupData() throws FailedLoginException {
-		this.currentStatus = this.fetchData(EndpointConstant.CURRENT_STATUS, CurrentStatus.class);
-		this.networkSetting = this.fetchData(EndpointConstant.NETWORK_SETTINGS, NetworkSetting.class);
-
-		String timeZone = Optional.ofNullable(this.networkSetting)
-				.map(NetworkSetting::getNtp).map(Ntp::getTimeZoneInfo).map(TimeZoneInfo::getName)
-				.orElse(null);
-		this.currentStatus.setTimeZone(timeZone);
+	private boolean isDataSetup() throws FailedLoginException {
+		try {
+			this.authenticate();
+			this.currentStatus = this.fetchData(EndpointConstant.CURRENT_STATUS, CurrentStatus.class);
+			this.networkSetting = this.fetchData(EndpointConstant.NETWORK_SETTINGS, NetworkSetting.class);
+
+			this.currentStatus.setTimeZone(Optional.ofNullable(this.networkSetting)
+					.map(NetworkSetting::getNtp).map(Ntp::getTimeZoneInfo).map(TimeZoneInfo::getName)
+					.orElse(null));
+			this.outlets = Optional.ofNullable(currentStatus)
+					.map(CurrentStatus::getDevices).filter(devices -> !devices.isEmpty())
+					.map(devices -> devices.get(0).getOutlets()).filter(outletList -> !outletList.isEmpty())
+					.orElse(new ArrayList<>());
+			this.groups = Optional.ofNullable(currentStatus)
+					.map(CurrentStatus::getDevices).filter(devices -> !devices.isEmpty())
+					.map(devices -> devices.get(0).getGroups()).filter(groupList -> !groupList.isEmpty())
+					.orElse(new ArrayList<>());
+
+			return true;
+		} catch (FailedLoginException e) {
+			throw e;
+		} catch (Exception e) {
+			this.logger.error(e.getMessage(), e);
```

Comment:
original exception's message already exists in original instance passed to the logger. Please provide a coherent message here, instead of e.getMessage() - what happens and when :)